provider/cluster_autoscaler_state.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type ClusterAutoscalerState struct {
+	Cluster                     types.String              `tfsdk:"cluster"`
+	BalanceSimilarNodeGroups    types.Bool                `tfsdk:"balance_similar_node_groups"`
+	SkipNodesWithLocalStorage   types.Bool                `tfsdk:"skip_nodes_with_local_storage"`
+	LogVerbosity                types.Int64               `tfsdk:"log_verbosity"`
+	MaxPodGracePeriod           types.Int64               `tfsdk:"max_pod_grace_period"`
+	PodPriorityThreshold        types.Int64               `tfsdk:"pod_priority_threshold"`
+	IgnoreDaemonsetsUtilization types.Bool                `tfsdk:"ignore_daemonsets_utilization"`
+	MaxNodeProvisionTime        types.String              `tfsdk:"max_node_provision_time"`
+	BalancingIgnoredLabels      types.List                `tfsdk:"balancing_ignored_labels"`
+	ResourceLimits              *AutoscalerResourceLimits `tfsdk:"resource_limits"`
+	ScaleDown                   *ScaleDownConfig          `tfsdk:"scale_down"`
+}
+
+type ResourceRange struct {
```

Comment:
Maybe we should add a `Autoscaler` prefix for these structs (like you did in `AutoscalerResourceLimits`)? To avoid confusion. I know we have a major refactoring soon and it might help.

Reply:
Sure, done!